packages/stream_chat_flutter/lib/src/message_input.dart

```diff
@@ -1988,8 +1989,6 @@ class MessageInputState extends State<MessageInput> {
       extraData: extraDataMap.isNotEmpty ? extraDataMap : null,
     );
 
-    _attachments[attachment.id] = attachment;
```

Comment:
I think we added the attachment at the start because the compression may take some time and we want to show it ASAP as the selected item in message input.

Reply:
I think there's a a better way to do that. If we want to do this, let's also add a check at the end checking for file sizes.